src/pal/src/include/pal/stackstring.hpp

```diff
@@ -44,34 +52,42 @@ class StackString
             return;
         }
 
+        if(preserve)
```

Comment:
I like the idea, but CopyMemeory is not going to be removed as we have the Stack as our primary datastore